include/common/thread.h

```diff
@@ -5,6 +5,40 @@
 #include <thread>
 
 namespace BoBRobotics {
+//! A simple RAII wrapper for std::thread
```

Comment:
Presumably this requires the thread to implement some sort of signalling to kill it before you join? I kinda feel the RAAIness just makes it easier to create a deadlock on exit